tests/e2e/multicluster/multicluster_primaryremote_test.go

```diff
@@ -0,0 +1,353 @@
+						Expect(kubectl.Apply(controlPlaneNamespace, exposeServiceYAML, kubeconfig)).To(Succeed(), "Expose Service creation failed on Primary Cluster")
+					})
+
+					It("updates Gateway status to Available", func(ctx SpecContext) {
+						Eventually((common.GetObject)).
+							WithArguments(ctx, clPrimary, kube.Key("istio-eastwestgateway", controlPlaneNamespace), &appsv1.Deployment{}).
+							Should(HaveCondition(appsv1.DeploymentAvailable, metav1.ConditionTrue), "Gateway is not Ready on Primary; unexpected Condition")
+					})
+				})
+
+				When("remote istio is created in Remote cluster", func() {
+					BeforeAll(func(ctx SpecContext) {
+						RemoteYAML := `
+apiVersion: sailoperator.io/v1alpha1
+kind: RemoteIstio
+metadata:
+  name: default
+spec:
+  version: %s
+  namespace: istio-system
+  values:
+    istiodRemote:
+      injectionPath: /inject/cluster/remote/net/network2
+    global:
+      remotePilotAddress: %s`
+
+						remotePilotAddress, err := common.GetSVCLoadBalancerAddress(ctx, clPrimary, controlPlaneNamespace, "istio-eastwestgateway")
+						Expect(remotePilotAddress).NotTo(BeEmpty(), "Remote Pilot Address is empty")
+						Expect(err).NotTo(HaveOccurred(), "Error getting Remote Pilot Address")
+						remoteIstioYAML := fmt.Sprintf(RemoteYAML, version.Name, remotePilotAddress)
+						Log("Remote Istio CR: ", remoteIstioYAML)
+						By("Creating Remote Istio CR on Remote Cluster")
+						Expect(kubectl.CreateFromString(remoteIstioYAML, kubeconfig2)).To(Succeed(), "Remote Istio Resource creation failed on Remote Cluster")
+
+						// Set the controlplane cluster and network for Remote namespace
+						By("Patching the istio-system namespace on Remote Cluster")
+						Expect(
+							kubectl.Patch("",
+								"namespace",
+								controlPlaneNamespace,
+								"merge",
+								`{"metadata":{"annotations":{"topology.istio.io/controlPlaneClusters":"cluster1"}}}`,
+								kubeconfig2)).
+							To(Succeed(), "Error patching istio-system namespace")
+						Expect(
+							kubectl.Patch("",
+								"namespace",
+								controlPlaneNamespace,
+								"merge",
+								`{"metadata":{"labels":{"topology.istio.io/network":"network2"}}}`,
+								kubeconfig2)).
+							To(Succeed(), "Error patching istio-system namespace")
+
+						// To be able to access the remote cluster from the primary cluster, we need to create a secret in the primary cluster
+						// RemoteIstio resource will not be Ready until the secret is created
+						// Get the internal IP of the control plane node in Remote cluster
+						internalIPRemote, err := kubectl.GetInternalIP("node-role.kubernetes.io/control-plane", kubeconfig2)
+						Expect(internalIPRemote).NotTo(BeEmpty(), "Internal IP is empty for Remote Cluster")
+						Expect(err).NotTo(HaveOccurred())
+
+						// Wait for the RemoteIstio CR to be created, this can be moved to a condition verification, but the resource it not will be Ready at this point
+						time.Sleep(5 * time.Second)
```

Comment:
Is this necessary? From the moment you create the `RemoteIstio` it should be created.

Reply:
I'm trying to find a way to wait for the resource because if I do not add the sleep it fails sometimes

Reply:
@fjglira What exactly do you need to wait for? What fails when you remove this sleep?

Reply:
Hey @nrfox and @luksa, for some reason, when you create the remote secret on Cluster 1 right after creating the `RemoteIstio` resource on Cluster 2, we get a `ReconcileError` and the test fails

Reply:
I'd say we should investigate this further but that doesn't have to block this PR.